pkg/services/ngalert/prom/query.go

```diff
@@ -43,7 +43,7 @@ func createQueryNode(datasourceUID, datasourceType, expr string, fromTimeRange,
 		RefID:         queryRefID,
 		RelativeTimeRange: models.RelativeTimeRange{
 			From: models.Duration(fromTimeRange + evaluationOffset),
-			To:   models.Duration(0 + evaluationOffset),
+			To:   models.Duration(evaluationOffset),
```

Comment:
The offset works backwards in time, e.g. if we ran a query at 00:30, then we run it with timestamp 00:29 ([see here](https://github.com/grafana/mimir/blob/1f0b1bdc5365f128a176988387785f6e1ba830ec/vendor/github.com/prometheus/prometheus/rules/alerting.go#L376))

```
			From: models.Duration(fromTimeRange - evaluationOffset),
			To:   models.Duration(-evaluationOffset),
```

Reply:
We use it in the `RelativeTimeRange` here and as I understand minuses are added when it gets converted to timerange:

https://github.com/grafana/grafana/blob/d3832c7f8bc0c1542b1841aa360aa359251daeee/pkg/services/ngalert/models/alert_query.go#L74-L79

This is also how the frontend sends it:

<img width="589" alt="image" src="https://github.com/user-attachments/assets/91020a45-b0c8-433b-ab79-f722ceb68d67" />

Reply:
Oh! Fascinating, well TIL.